Src/ILGPU.Algorithms/Runtime/Cuda/CuFFTPlan.cs

```diff
@@ -0,0 +1,390 @@
+﻿// ---------------------------------------------------------------------------------------
+//                                   ILGPU Algorithms
+//                           Copyright (c) 2021 ILGPU Project
+//                                    www.ilgpu.net
+//
+// File: CuFFTPlan.cs
+//
+// This file is part of ILGPU and is distributed under the University of Illinois Open
+// Source License. See LICENSE.txt for details.
+// ---------------------------------------------------------------------------------------
+
+using ILGPU.Runtime.Cuda.API;
+using ILGPU.Util;
+using System;
+
+namespace ILGPU.Runtime.Cuda
+{
+    /// <summary>
+    /// Represents a cuFFT plan.
+    /// </summary>
+    [CLSCompliant(false)]
+    public sealed partial class CuFFTPlan : DisposeBase
+    {
+        /// <summary>
+        /// The underlying API wrapper.
+        /// </summary>
+        public CuFFTAPI API { get; }
```

Comment:
Nit: we usually place properties below the constructor

Reply:
Done.